app/api/jobs.py

```diff
@@ -0,0 +1,91 @@
+from fastapi import APIRouter, Request, HTTPException, Depends, Response
+from ..db import get_database, get_JobImage_path, get_export_path
+from ..models import JobItem, JobItemPayload, AccessTokenPayload
+from app.utils.validation import validate_image_file_type, validate_uuid
+from ..auth_helpers import authorize, authorize_admin
+from fastapi.datastructures import UploadFile
+from fastapi.param_functions import File
+import os
+import shutil
+from datetime import datetime
+from uuid import UUID, uuid4
+from starlette.responses import FileResponse
+
+
+router = APIRouter()
+
+
+@router.get('/')
+def get_jobs(request: Request):
+    db = get_database(request)
+    jobs = db.jobs.find()
+    return [JobItem.parse_obj(job) for job in jobs]
+
+
+@router.get('/{id}')
+def get_jobs(request: Request, id: str):
+    db = get_database(request)
+    job = db.jobs.find_one({'id': UUID(id)})
+    if job == None:
+        raise HTTPException(404, "No such job with this id")
+    return JobItem.parse_obj(job)
+
+
+@router.post('/')
+def create_job(request: Request, job: JobItemPayload, token: AccessTokenPayload = Depends(authorize_admin)):
+
+    db = get_database(request)
+    item = job.dict()
+    item['id'] = uuid4()
+    item['published_date'] = datetime.now()
+    _job = JobItem.parse_obj(item)
+    db.jobs.insert_one(_job.dict())
+    return {'id': _job.id}
+
+
+@router.delete('/{id}')
+def delete_job(request: Request, id: str, token: AccessTokenPayload = Depends(authorize_admin)):
+    db = get_database(request)
+    job = db.jobs.find_one({'id': UUID(id)})
+    if not job:
+        raise HTTPException(400, "Job could not be found")
+    db.jobs.delete_one({'id': job['id']})
+    return {'id': id}
+
+
+@router.put('/{id}')
+def update_job(request: Request, id: str, job: JobItemPayload, token: AccessTokenPayload = Depends(authorize_admin)):
+    db = get_database(request)
+    original_job = db.jobs.find_one({'id': UUID(id)})
+    if not original_job:
+        raise HTTPException(400, "Job could not be found")
+    db.jobs.update_one({'id': UUID(id)}, {
```

Comment:
I think there is find_one_and_update so that we can avoid find_one and then do delete_one later